cider/exec/module/CiderRuntimeModule.cpp

```diff
@@ -744,14 +744,32 @@ CiderBatch CiderRuntimeModule::setSchemaAndUpdateCountDistinctResIfNeed(
       int flatten_index = schema->getFlattenColIndex(i);
       if (schema->getColHints()[i] == ColumnHint::PartialAVG &&
           target_infos[flatten_index].agg_kind == SQLAgg::kSUM &&
-          target_infos[flatten_index].agg_arg_type.is_integer() &&
           generator::getSQLTypeInfo(schema->getColumnTypeById(i).struct_().types(0))
                   .get_type() == SQLTypes::kDOUBLE) {
-        int64_t* result = const_cast<int64_t*>(
-            reinterpret_cast<const int64_t*>(outBuffers[flatten_index]));
-        double value = (double)result[0];
-        double* cast_buffer = reinterpret_cast<double*>(result);
-        cast_buffer[0] = value;
+        if (target_infos[flatten_index].agg_arg_type.is_integer()) {
+          int64_t* result = const_cast<int64_t*>(
+              reinterpret_cast<const int64_t*>(outBuffers[flatten_index]));
+          double* cast_buffer = reinterpret_cast<double*>(result);
+          switch (result[0]) {
+            case std::numeric_limits<int8_t>::min():
+            case std::numeric_limits<int16_t>::min():
+            case std::numeric_limits<int32_t>::min():
+            case std::numeric_limits<int64_t>::min():
+              cast_buffer[0] = NULL_VALUE_DOUBLE;
+              break;
+            default:
+              cast_buffer[0] = (double)result[0];
+          }
+        } else if (target_infos[flatten_index].agg_arg_type.is_fp()) {
+          double* result = const_cast<double*>(
+              reinterpret_cast<const double*>(outBuffers[flatten_index]));
+          double* cast_buffer = reinterpret_cast<double*>(result);
+          if (result[0] == std::numeric_limits<float>::min()) {
+            cast_buffer[0] = NULL_VALUE_DOUBLE;
```

Comment:
Similar as above, if `agg_arg_type.is_fp()`, the null value could only be `NULL_VALUE_DOUBLE` and no other cases and for this case, you should modify `[extract_column](https://github.com/intel/BDTK/blob/4f316da4e36b8646cbdf3add8d13b124ed4c94d1/cider/exec/module/CiderRuntimeModule.cpp#L390)`.

Reply:
Get it, let me modify `extract_column` to transfer back null values of float.